toolkit/basemapgallery/src/main/java/com/arcgismaps/toolkit/basemapgallery/BasemapGallery.kt

```diff
@@ -0,0 +1,114 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package com.arcgismaps.toolkit.basemapgallery
+
+import android.util.Log
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.lazy.grid.GridCells
+import androidx.compose.foundation.lazy.grid.LazyVerticalGrid
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.asImageBitmap
+import androidx.compose.ui.graphics.painter.BitmapPainter
+import androidx.compose.ui.graphics.painter.Painter
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+
+/**
+ * Produces the composable parts required to compose a [BasemapGalleryItem] in a [BasemapGallery].
+ *
+ * @param basemapGalleryItem the [BasemapGalleryItem]
+ */
+@Composable
+internal fun BasemapGalleryItem(basemapGalleryItem: BasemapGalleryItem) {
+    val placeholder = painterResource(R.drawable.basemap)
+    val thumbnail: MutableState<Painter> = remember { mutableStateOf(placeholder) }
+    LaunchedEffect(thumbnail) {
+        basemapGalleryItem.thumbnailProvider()?.let { bitmap ->
+            bitmap.asImageBitmap().let { imageBitmap ->
+                thumbnail.value = BitmapPainter(imageBitmap)
+            }
+        }
+        thumbnail
+    }
+
+    Image(
+        painter = thumbnail.value,
+        contentDescription = basemapGalleryItem.title,
+        modifier = Modifier.clip(RoundedCornerShape(8.dp))
+    )
+    Text(text = basemapGalleryItem.title, textAlign = TextAlign.Center)
+}
+
+/**
+ * A gallery of [BasemapGalleryItem]s.
+ *
+ * @param basemapGalleryItems the items to show in the gallery
+ * @param modifier the modifier to apply to this gallery
+ * @param onItemClick a lambda to execute when a gallery item is clicked
+ */
+@Composable
+public fun BasemapGallery(
+    basemapGalleryItems: List<BasemapGalleryItem>,
+    modifier: Modifier = Modifier,
+    onItemClick: (BasemapGalleryItem) -> Unit
```

Comment:
since this is not a default parameter, it needs to go above `modifier`:
```suggestion
    onItemClick: (BasemapGalleryItem) -> Unit,
    modifier: Modifier = Modifier,
```

The API ref doc would also need to be updated.

Reply:
I wonder if it would be better to default it to `onItemClick: (BasemapGalleryItem) -> Unit = {}` so it could still be the last parameter to allow a trailing lambda?

Reply:
A default parameter would only make sense here if this lambda was optional. But I guess a user should always handle the `onItemClick`, otherwise the gallery would not be able to fullfil its purpose? This is similar to [Button](https://developer.android.com/reference/kotlin/androidx/compose/material3/package-summary#Button(kotlin.Function0,androidx.compose.ui.Modifier,kotlin.Boolean,androidx.compose.ui.graphics.Shape,androidx.compose.material3.ButtonColors,androidx.compose.material3.ButtonElevation,androidx.compose.foundation.BorderStroke,androidx.compose.foundation.layout.PaddingValues,androidx.compose.foundation.interaction.MutableInteractionSource,kotlin.Function1)), which has a mndatory `onClick` parameter.

Reply:
Reordered parameters